waltz-ng/client/auth-sources/components/non-auth-sources-list.js

```diff
@@ -0,0 +1,69 @@
+import _ from "lodash";
+import {initialiseData} from "../../common";
+
+const bindings = {
+    dataTypes: '<',
+    distributors: '<',
+    consumers: '<',
+};
+
+
+const initialState = {
+    nonAuthSources: [],
+    distributors: [],
+};
+
+
+const template = require('./non-auth-sources-list.html');
+
+
+function calculate(dataTypes = [], distributorsByDataType = [], allConsumers = []) {
+    const consumersBySource = _.groupBy(allConsumers, 'source.id');
+    const dataTypesById = _.keyBy(dataTypes, 'id');
+
+    const distributersFlattened = _.flatMap(distributorsByDataType, (values, key) => {
+        const dataType = dataTypesById[key] || {};
+        return  _.map(values, v => Object.assign(v, {dataType} ));
+    } );
+
+    const nonAuthSources = _.chain(distributersFlattened)
+        .map(distributor => {
+            const consumers = consumersBySource[distributor.id] || [];
+            return Object.assign({}, distributor, {consumers});
+        })
+        .value();
```

Comment:
actually - looking further you could chain from the line 24 thru' to completion....